crates/router/src/compatibility/stripe/customers.rs

```diff
@@ -167,11 +167,11 @@ pub async fn customer_delete(
 pub async fn list_customer_payment_method_api(
     state: web::Data<routes::AppState>,
     req: HttpRequest,
-    path: web::Path<String>,
+    json_payload: web::Query<payment_methods::PaymentMethodListRequest>,
```

Comment:
you need to use the customer_id here. 
For this PR use case create separate API endpoint don't change existing one.